kikuchipy/signals/ebsd_master_pattern.py

```diff
@@ -61,6 +74,145 @@ def __init__(self, *args, **kwargs):
         self.projection = kwargs.pop("projection", None)
         self.hemisphere = kwargs.pop("hemisphere", None)
 
+    def get_patterns(
+        self,
+        rotations: Rotation,
+        detector: EBSDDetector,
+        energy: int,
+        n_chunk: Optional[int] = -1,
+        dtype_out: Optional[np.dtype] = np.float32,
+        compute: Optional[bool] = False,
+    ) -> Union[EBSD, LazyEBSD]:
+        """Creates a dictionary of EBSD patterns for a sample in the
+        (RD, TD, ND) reference frame, given a set of local crystal lattice
+        rotations and a detector model from a master pattern in the Lambert
+        projection.
+
+        Parameters
+        ----------
+        rotations : Rotation
+            Set of unit cell rotations to get patterns from.
+        detector : EBSDDetector
+            EBSDDetector object describing the detector geometry with one
+            projection center.
+        energy : int
+            The wanted energy in the master pattern.
+        n_chunk : int, optional
+            The number of chunks the data should be split up into. By default,
+            this is set so each chunk is around 100 MB.
+        dtype_out : numpy.dtype, optional
+            Data type of the returned patterns, by default np.float32.
+        compute : bool, optional
+            Wheter or not the dask.compute() function should be called, by
+            default false.
+            For more information visit the dask documentation at:
+            https://docs.dask.org/en/latest/api.html#dask.compute
+
+        Returns
+        ----------
+        this : EBSD or LazyEBSD
+            All the simulated EBSD patterns with the shape (number of rotations,
+            detector pixels in x direction, detector pixels in y direction).
+
+        Notes
+        ----------
+        If the master pattern phase has a non-centrosymmetric point group, both
+        the northern and southern hemispheres must be provided.
+        For more details regarding the reference frame visit the reference frame
+        user guide at: https://kikuchipy.org/en/latest/reference_frames.html.
+        """
+
+        if self.projection != "lambert":
+            raise NotImplementedError(
+                "Method only supports master patterns in lambert projection!"
+            )
+        pc = detector.pc_emsoft()
+        if len(pc) > 1:
+            raise ValueError("Method only supports a single projection center!")
+
+        # 4 cases
+        # Has energies, has hemis - Case 1
+        if len(self.axes_manager.shape) == 4:
+            energies = self.axes_manager["energy"].axis
+            energy_index = (np.abs(energies - energy)).argmin()
+            mpn = self.data[0, energy_index]
+            mps = self.data[1, energy_index]
+        # no energies, no hemis - Case 2
+        elif len(self.axes_manager.shape) == 2:
+            if not self.phase.point_group.contains_inversion:
+                raise HemisphereError(
```

Comment:
What is the reason for not just using a ValueError here (as is done above for the number of PCs)?

Reply:
It could be a ValueError here, but if it raises a ValueError in the Try/Except L150:L162 it has to catch too many things, unless I would make it a nested try/except maybe? And since they are the same error I thought it would be less confusing if they had the same name across the code.

Reply:
The way I see it:
* We shouldn't be able to sample a master pattern without knowing whether its point group is centrosymmetric or not. Thus the master pattern object MUST have a valid `phase` property, and the check for centrosymmetry can be outside the try/except in the next else clause.
* Then the try/except for whether the master pattern signal has an energy axis can be as below.

Reply:
I guess the `AttributeError` is more fitting: https://docs.python.org/3/library/exceptions.html#AttributeError

Reply:
Changed my mind, I would consider this an `AttributeError`, i.e. there is something incompatible with the object (phase attribute (point group) requires a southern hemisphere, which isn't present in the data attribute). I think this with the error message is sufficiently clear for the user.